### ISSUE TITLE
[Bug Report] Unable to attach label to input when using <el-form-item> label attribute (WCAG 4.1.2)

### ISSUE THREAD

**Issue by Giwayume** [element-plus/element-plus#1846]
### Element Plus version
1.0.2-beta.40

### OS/Browsers version
All

### Vue version
3.0.11

### Reproduction Link
https://element-plus.org/#/en-US/component/form

### Steps to reproduce
Create a simple el-form-item with label attribute and an input inside.
```
<el-form-item label="Activity name">
    <el-input></el-input>
</el-form-item>
```

### What is Expected?
Markup resembling this is generated:
```
<label for="input_id">Activity name</label>
<input id="input_id">
```

Either:
1. The "for" attribute on the generated label as well as the "id" attribute on the input is automatically generated in this scenario ([Bootstrap Vue does this...](https://bootstrap-vue.org/docs/components/form/#form)). And a similar story for other types of inputs like el-select, el-radio-group, etc.
2. Expose a new property called "label-for" or similar so the label can be programmatically attached to the input and follow web accessibility guidelines [WCAG 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html).

### What is actually happening?
There seems to be no way to attach the form label to the associated inputs it is labeling short of hacking it by writing a directive to do so.

**Comment by JeremyWuuuuu** [comment]
We kind of having the same API named `for` in `form-item` you can use that.
But I think you made your point that we should generate form item with `id` while the `for` was given.

**Comment by JeremyWuuuuu** [comment]
Also according to the documentation site you provided, you are still going to provide the id to form items on you own
```vue
 <b-form-group
        id="input-group-1"
        label="Email address:"
        label-for="input-1"
        description="We'll never share your email with anyone else."
      >
        <b-form-input
          id="input-1"
          v-model="form.email"
          type="email"
          placeholder="Enter email"
          required
        ></b-form-input>
```

**Comment by Giwayume** [comment]
@JeremyWuuuuu ok, I don't see that in the English documentation for form item, at least. So maybe this issue is just update the documentation.

![image](https://user-images.githubusercontent.com/4075314/114883382-df4d2500-9dd2-11eb-87f2-716039818cee.png)

And yes, Bootstrap Vue allows you to define it manually, but if you leave it out they auto-generate it which is very helpful.

**Comment by Giwayume** [comment]
Actually I could be remembering wrong, if you leave "label-for" out they generate a fieldset/legend combination instead, which doesn't require the for...id relationship. https://jsfiddle.net/mh7nbqy0/

Not saying el-form-item needs to do that, but that does bring up an interesting point in documentation examples where multiple inputs are provided, fieldset/legend + aria-label on each input is the correct markup to use. Such as this:

![image](https://user-images.githubusercontent.com/4075314/114899495-01e63a80-9de1-11eb-8f1b-60aecd212cd7.png)

This markup would associate the overall "Activity time" label with both inputs, as well as label each input individually.

```
<fieldset class="el-form-item is-required">
	<legend class="el-form-item__label">Activity time</legend>
	<div class="el-form-item__content">
		<div class="el-col el-col-11">
			<div class="el-form-item is-required">
				<div class="el-form-item__content">
					<div class="el-input">
						<input aria-label="Date" class="el-input__inner" placeholder="Pick a date">
					</div>
				</div>
			</div>
		</div>
		<div class="el-col el-col-11">
			<div class="el-form-item is-required">
				<div class="el-form-item__content">
					<div class="el-input">
						<input aria-label="Time" class="el-input__inner" placeholder="Pick a time">
					</div>
				</div>
			</div>
		</div>
	</div>
</fieldset>
```

**Comment by JeremyWuuuuu** [comment]
Yeah, I think you quite made the point, we should consider implementing that. We also need to update the documentation, this is serious issue that we didn't mention that in the documentation, could've been the legacy issue we have in `ElementUI`. I will make sure that we add the missing attributes in our next release.

**Comment by JeremyWuuuuu** [comment]
About the implementation Bootstrap had, I think we can learn from them to create better usability for our users also less mental burden, that could be an awesome improvement.
